### DISCUSSION TITLE
Could elevation work on windows 10 with self-contained WASDK?

### DISCUSSION THREAD

**Discussion by roxk** [microsoft/WindowsAppSDK#2378] [Dev team discussions]
Quoting from [this discussion](https://github.com/microsoft/WindowsAppSDK/discussions/671):

> actually WinUI 3 in packaged apps also does NOT support elevation. It's an OS-level issue, COM/WinRT doesn't allow per-user class registrations to be activated from an elevated process. Inbox APIs work when elevated, but WinAppSDK isn't inbox. We've fixed that in the OS in the next release.

And according to [1.1-preview1's discussion](https://github.com/microsoft/WindowsAppSDK/discussions/2331#discussioncomment-2466454), it seems it is decided that WASDK 1.1 will not support elevation on windows 10.

I noticed that terminal is able to support elevation by [using SxS manifest](https://github.com/microsoft/WindowsAppSDK/discussions/671#discussioncomment-1093194). Couldn't WASDK at least support elevation on windows 10 for self-contained deployment? Self-contained WASDK already needs SxS for RegFreeWinRT anyways.

I have high hope for WASDK and I'm afraid the down-level/polyfill promises failing one after another would hurt its reputation. Hope this can start some discussion on potential solutions that can bring elevation support to windows 10.

---

Is there any plan to bring the fixes for WinRT/COM activation from elevated process (which is already in windows 11) to future windows 10 update, e.g. 22H1/22H2? This would allow packaged app to have elevation support in windows 10 as well.

**Comment by roxk** [comment]
Sorry for self-pushing, but the thing that I was afraid of is showing up already: [see this tweet complaining about how WASDK broke the down-level poly-fill promise](https://twitter.com/dotMorten/status/1521969187253485568). Is there any plan regarding elevation support on windows 10? Even just SxS like how terminal achieved elevation with custom WinRT component? This is currently biting [power toys](https://github.com/microsoft/PowerToys/issues/18019) apparently.

@DrusTheAxe Not sure if it's appropriate to ping directly but here I'm pinging, as IMO this is very urgent (affecting power toy) and any kind of reply is appreciated. Thanks!

**Reply by AdamBraden** [reply]
Elevated support for COM & WinRT has a lot of implementation nuances when it comes to enabling downlevel to RS5, and to support MSIX packaged apps, unpackaged apps, and self-contained apps. The SxS manifest is only a partial solution. The support in 1.1 is to reconcile that across all scenarios. 

However there is a detail we couldn't solve in the WinAppSDK and that requires an OS servicing update. Those OS updates are available for download at these links. They will be part of the next patch Tuesday and thus be broadly available for your customers soon:
- Windows 11 - [April 25, 2022—KB5012643 (OS Build 22000.652) Preview](https://support.microsoft.com/en-us/topic/april-25-2022-kb5012643-os-build-22000-652-preview-43a75ee7-d857-4943-a2b9-f961538bd2b0)
- Windows 10 - [April 25, 2022—KB5011831 (OS Builds 19042.1682, 19043.1682, and 19044.1682) Preview](https://support.microsoft.com/en-us/topic/april-25-2022-kb5011831-os-builds-19042-1682-19043-1682-and-19044-1682-preview-fe4ff411-d25a-4185-aabb-8bc66e9dbb6c)

- Note that the OS fix for RS5 is still pending and will come in a future update.
